crates/ibc/src/core/ics24_host/identifier.rs

```diff
@@ -412,9 +412,8 @@ impl Default for PortId {
     feature = "borsh",
     derive(borsh::BorshSerialize, borsh::BorshDeserialize)
 )]
-#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct ChannelId(String);
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct ChannelId(u64);
```

Comment:
Doesn't this break compatibility with other IBC chains? I don't think the spec requires that the channel ID be a `u64`, only that it's an `Identifier`, [which is defined as](https://github.com/cosmos/ibc/tree/main/spec/core/ics-024-host-requirements#paths-identifiers-separators)

> An `Identifier` is a bytestring used as a key for an object stored in state, such as a connection, channel, or light client.
>
> Identifiers MUST be non-empty (of positive integer length).
> 
> Identifiers MUST consist of characters in one of the following categories only:
> - Alphanumeric
> - `.`, `_`, `+`, `-`, `#`
> - `[`, `]`, `<`, `>`

Reply:
yes indeed, I did some digging into the history of this issue. It was created when `ChannelId` was a `u64`. However we changed it to `String` in [this PR](https://github.com/informalsystems/hermes/pull/2361), for exactly these reasons. I will close this PR and the issue as "won't do".

Thank you @DaviRain-Su for ultimately bringing this to light; sorry for needing to throw away all this code.